maint_tools/check_gha_workflow.py

```diff
@@ -0,0 +1,263 @@
+"""Collects and plot run time of jobs in a GHA workflow.
+
+Requires:
+- requests
+- rich
+- pandas
+- plotly
+- kaleido
+
+For a given github action workflow:
+- ping the github API to collect the start and end time
+  of all the jobs of the different runs of that workflow,
+- saves to TSV and plots the duration of each job against time.
+
+This script should in principle run for any repo and any workflow.
+
+This may require a github token to function
+that you can pass to the script
+or save in a file the script will read.
+
+USAGE
+-----
+
+.. code-block:: bash
+
+    python maint_tools/check_gha_workflow.py
+
+    # or by passing github token directly
+    python maint_tools/check_gha_workflow.py GITHUB_TOKEN
+
+
+"""
+
+import sys
+import warnings
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+import plotly.express as px
+import requests
+from rich import print
+
+USER = "nilearn"
+REPO = "nilearn"
+
+# Your github username
+USERNAME = "Remi-Gau"
```

Comment:
Is this something that we have to set for you individually, or could it be set under the org ?

Reply:
The username and the token file are there if you want or need to run the script locally.
In CI we use the org token directly.

I will improve the comment to better explain this.